node-commons/src/main/java/org/apache/iotdb/commons/utils/BasicStructureSerDeUtil.java

```diff
@@ -239,4 +241,78 @@ public static Map<Integer, List<Integer>> readIntMapLists(ByteBuffer buffer) {
     }
     return map;
   }
+
+  /**
+   * write int set to buffer
+   *
+   * @param set Set<Integer> set
+   * @param buffer
+   * @return length
+   */
+  public static int writeIntSet(Set<Integer> set, ByteBuffer buffer) {
+    if (set == null) {
+      return write(-1, buffer);
+    }
+    int length = 0;
+    buffer.putInt(set.size());
+    for (Integer i : set) {
+      buffer.putInt(i);
+    }
+    return length;
+  }
+
+  /**
+   * read int set from buffer
+   *
+   * @param buffer
+   * @return Set<Integer>
+   */
+  public static Set<Integer> readIntSet(ByteBuffer buffer) {
+    int length = readInt(buffer);
+    if (length == -1) {
+      return null;
+    }
+    Set<Integer> set = new HashSet<>();
+    for (int i = 0; i < length; i++) {
+      set.add(readInt(buffer));
+    }
+    return set;
+  }
+
+  /**
+   * write string list to buffer
+   *
+   * @param strings List<String> strings
+   * @param buffer
+   * @return length
+   */
+  public static int writeStringList(List<String> strings, ByteBuffer buffer) {
+    if (strings == null) {
+      return write(-1, buffer);
+    }
+    int length = 0;
+    buffer.putInt(strings.size());
+    for (String s : strings) {
+      write(s, buffer);
+    }
+    return length;
+  }
+
+  /**
+   * read string list to buffer
+   *
+   * @param buffer
+   * @return List<String>
+   */
+  public static List<String> readStringList(ByteBuffer buffer) {
+    int length = readInt(buffer);
+    if (length == -1) {
+      return null;
+    }
+    List<String> strings = new ArrayList<>();
+    for (int i = 0; i < length; i++) {
+      strings.add(readString(buffer));
+    }
+    return strings;
+  }
```

Comment:
Please add tests in BasicStructureSerDeUtilTest.

Reply:
fix